syncer/syncer.go

```diff
@@ -2451,7 +2446,7 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext, o
 
 		// DDL is sequentially synchronized in this syncer's main process goroutine
 		// ignore DDL that is older or same as table checkpoint, to avoid sync again for already synced DDLs
-		if s.checkpoint.IsOlderThanTablePoint(tableNames[0][0].Schema, tableNames[0][0].Name, *ec.currentLocation, true) {
+		if s.checkpoint.IsOlderThanTablePoint(tableNames[0][0], *ec.currentLocation, true) {
```

Comment:
`tableNames` should be tables, so please change `sqlDDL, tableNames, stmt, handleErr := s.routeDDL(parser.New(), usedSchema, sql)` to  `sqlDDL, tables, stmt, handleErr := s.routeDDL(parser.New(), usedSchema, sql)` in L2437

Reply:
fix in https://github.com/pingcap/dm/pull/2106/commits/d6ec42afcecd3d85c8f76815b8850284761d7a6d